examples/hosts/app-integration/external-data/src/view/appView.tsx

```diff
@@ -25,7 +25,7 @@ export const AppView: React.FC<IAppViewProps> = (props: IAppViewProps) => {
 	const { model } = props;
 	const taskList = model.baseDocument.getTaskList("task-list-1");
 	return taskList !== undefined ? (
-		<TaskListView taskList={taskList} />
+		<TaskListView taskList={taskList} model={model} />
```

Comment:
Is there a need to pass in the whole model here? In the same that taskList is retrieved just prior to being passed into the TaskListView component, the leader property can also be passed in here as `leader = {model.leader}`.

Reply:
@WayneFerrao bringing back this topic, I'm still uneasy with passing the whole model down. The model object itself never changes after it gets passed to the AppView, I think, so I'm not sure what would cause things to re-render when a client claims leadership by calling a method on model, which does not impact any React state. If you have two clients (browser tabs) connected to the same container, do you see the UI change in both if you claim leadership in one, then the other, then the first one again, etc?

Reply:
I felt like it was appropriate to access the various methods within, but it's true that they can just as easily be accessed by passing them in individually. I'll make the change.
When I have two clients, the state does change for the first, then the second. But it does not change for the first, when the 2nd client claims leadership; it does not go back to the "changes in progress view"

Reply:
Yeah, I think we're missing something then because that should happen. The missing piece might be that we want to have the id of the current leader as React state somewhere close to the top of the component hierarchy, passed down as a prop, and the component that holds it as state should monitor for when it changes (i.e. when there's a change in the root SharedMap that holds the id of the leader) so it can re-render in response to that and cause the children to re-render as well, so the UI gets updated. 

Maybe something like this:

- `BaseDocument`'s `setLeader()` method emits an event with the new leader id (similar to what `addTaskList()` does).
- The `AppView` component is the one that holds the leader id in its React state, and sets up an event to listen for the new event from `BaseDocument` (since it has it in `model.baseDocument`).